src/chemistry/aerosol/cloud_aqueous_chemistry.F90

```diff
@@ -0,0 +1,894 @@
+    use spmd_utils,   only : masterproc
+    use phys_control, only : phys_getopts
+    use carma_flags_mod, only : carma_do_cloudborne
+    use clouds, only : sox_cldaero_init
+
+    logical :: modal_aerosols
+
+    call phys_getopts( prog_modal_aero_out=modal_aerosols )
+    cloud_borne = modal_aerosols .or. carma_do_cloudborne
+
+    !-----------------------------------------------------------------
+    !       ... get species indicies
+    !-----------------------------------------------------------------
+
+    if (cloud_borne) then
+       id_h2so4 = get_spc_ndx( 'H2SO4' )
+    else
+       id_so4 = get_spc_ndx( 'SO4' )
+    endif
+    id_msa = get_spc_ndx( 'MSA' )
+
+    inv_so2 = .false.
+    id_so2 = get_inv_ndx( 'SO2' )
+    inv_so2 = id_so2 > 0
+    if ( .not. inv_so2 ) then
+       id_so2 = get_spc_ndx( 'SO2' )
+    endif
+
+    inv_NH3 = .false.
+    id_NH3 = get_inv_ndx( 'NH3' )
+    inv_NH3 = id_NH3 > 0
+    if ( .not. inv_NH3 ) then
+       id_NH3 = get_spc_ndx( 'NH3' )
+    endif
+
+    inv_HNO3 = .false.
+    id_HNO3 = get_inv_ndx( 'HNO3' )
+    inv_HNO3 = id_hno3 > 0
+    if ( .not. inv_HNO3 ) then
+       id_HNO3 = get_spc_ndx( 'HNO3' )
+    endif
+
+    inv_H2O2 = .false.
+    id_H2O2 = get_inv_ndx( 'H2O2' )
+    inv_H2O2 = id_H2O2 > 0
+    if ( .not. inv_H2O2 ) then
+       id_H2O2 = get_spc_ndx( 'H2O2' )
+    endif
+
+    inv_HO2 = .false.
+    id_HO2 = get_inv_ndx( 'HO2' )
+    inv_HO2 = id_HO2 > 0
+    if ( .not. inv_HO2 ) then
+       id_HO2 = get_spc_ndx( 'HO2' )
+    endif
+
+    inv_o3 = get_inv_ndx( 'O3' ) > 0
+    if (inv_o3) then
+       id_o3 = get_inv_ndx( 'O3' )
+    else
+       id_o3 = get_spc_ndx( 'O3' )
+    endif
+    inv_ho2 = get_inv_ndx( 'HO2' ) > 0
+    if (inv_ho2) then
+       id_ho2 = get_inv_ndx( 'HO2' )
+    else
+       id_ho2 = get_spc_ndx( 'HO2' )
+    endif
+
+    has_sox = (id_so2>0) .and. (id_h2o2>0) .and. (id_o3>0) .and. (id_ho2>0)
+    if (cloud_borne) then
+       has_sox = has_sox .and. (id_h2so4>0)
+    else
+       has_sox = has_sox .and. (id_so4>0) .and. (id_nh3>0)
+    endif
+
+    if (masterproc) then
+       write(iulog,*) 'sox_inti: has_sox = ',has_sox
+    endif
+
+    if( has_sox ) then
+       if (masterproc) then
+          write(iulog,*) '-----------------------------------------'
+          write(iulog,*) ' mo_setsox will do sox aerosols'
+          write(iulog,*) '-----------------------------------------'
+       endif
+    else
+       if (masterproc) then
+          write(iulog,*) '-----------------------------------------'
+          write(iulog,*) ' mo_setsox will not do sox aerosols'
+          write(iulog,*) '-----------------------------------------'
+       endif
+       return
+    end if
+
+    call sox_cldaero_init()
+
+  end subroutine sox_inti
+
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+  subroutine setsox( state, &
```

Comment:
@tilmes @marybarthbrock - could we get a brief (2-3 sentence) high-level description of what this function is doing in science terms? We should also rename the function to be more descriptive (e.g., `solve_cloud_aqueous_chemistry()` or similar)

Reply:
Maybe we can consider using Doxygen style comments

Reply:
For me, a very brief description of setsox will be
"This subroutine calculates the formation of sulfate and updates the sulfate concentration due to cloud aqueous chemistry (i.e., updating qin / qcw). This subroutine also outputs the production rates (in kg/m2/s) of various reactions of sulfur species with various oxidants (i.e., outputting things like aqso4_h2o2 and aqh2so4)."
Something along the line... Mary and Simone et al. can add onto this...